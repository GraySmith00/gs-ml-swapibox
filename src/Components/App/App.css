.app {
  margin: 0 auto;
  font-family: 'Pathway Gothic One', sans-serif; }

.header {
  background-color: rgba(0, 0, 0, 0.5);
  padding: 20px 0 20px 0;
  position: fixed;
  top: 0;
  width: 100%;
  font-size: 1.5em;
  color: #fff; }

.header-container,
.main-container {
  display: flex;
  justify-content: space-between;
  align-items: center;
  width: 80%;
  margin: 0 auto; }

.brand-image {
  height: 1em; }

.nav-btns {
  display: flex;
  justify-content: space-between;
  width: 50%; }

.nav-link {
  text-decoration: none;
  margin-left: 1em; }

.App-title {
  font-size: 1.5em; }

.App-intro {
  font-size: large; }

.nav-link {
  color: #fff; }

@media screen and (max-width: 900px) {
  .header-container,
  .main-container {
    width: 90%; } }

@media screen and (max-width: 815px) {
  .header-container {
    flex-direction: column;
    width: 95%; }
  .brand-image {
    margin-bottom: 1rem; }
  .nav-btns {
    width: 75%; } }

@media screen and (max-width: 515px) {
  .nav-btns {
    width: 95%; }
  .nav-link {
    margin-left: 0; } }
