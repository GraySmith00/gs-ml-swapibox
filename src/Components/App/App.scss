.app {
  margin: 0 auto;
  font-family: 'Pathway Gothic One', sans-serif;
}

.main-container {
  display: flex;
  justify-content: space-between;
  align-items: center;

  width: 80%;
  margin: 0 auto;
}

.App-title {
  font-size: 1.5em;
}

.App-intro {
  font-size: large;
}

@media screen and (max-width: 900px) {
  .main-container {
    width: 90%;
  }
}
