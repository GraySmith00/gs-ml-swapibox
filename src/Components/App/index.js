import React, { Component } from 'react';
import {
  BrowserRouter as Router,
  Route,
  NavLink,
  Switch
} from 'react-router-dom';

import ScrollText from '../ScrollText';
import Landing from '../Landing';
import CategoryContainer from '../CategoryContainer';

import { initialFetchCall } from '../../helpers';

import './App.css';

class App extends Component {
  constructor() {
    super();
    this.state = {
      favorites: [],
      peopleData: [],
      planetsData: [],
      vehiclesData: [],
      errors: null
    };
  }

  setCategoryState = async category => {
    if (this.state[`${category}Data`].length === 0) {
      try {
        const data = await initialFetchCall(category);
        this.setState({ [`${category}Data`]: data });
      } catch (error) {
        this.setState({ errors: 'failed' });
      }
    }
  };

  toggleFavorite = item => {
    const alreadyFavorite = this.state.favorites.find(
      favorite => favorite.name === item.name
    );
    if (alreadyFavorite) {
      const newFavorites = this.state.favorites.filter(
        favorite => favorite.name !== item.name
      );
      this.setState({
        favorites: newFavorites
      });
    } else {
      const newFavorites = [...this.state.favorites, item];
      this.setState({
        favorites: newFavorites
      });
    }
  };

  render() {
    const { peopleData, planetsData, vehiclesData } = this.state;
    return (
      <div className="App">
        <Router>
          <div>
            <header className="App-header">
              <h1 className="App-title">SWAPIbox</h1>
              <nav className="nav-btns">
                <NavLink exact to="/people" className="nav-link">
                  People
                </NavLink>
                <NavLink exact to="/planets" className="nav-link">
                  Planets
                </NavLink>
                <NavLink exact to="/vehicles" className="nav-link">
                  Vehicles
                </NavLink>
              </nav>
            </header>
            <main>
              <Switch>
                <Route exact path="/" component={Landing} />
                <Route
                  path="/people"
                  render={() => {
                    this.setCategoryState('people');
                    return (
                      <CategoryContainer
                        toggleFavorite={this.toggleFavorite}
                        currentData={peopleData}
                      />
                    );
                  }}
                />
                <Route
                  exact
                  path="/planets"
                  render={() => {
                    this.setCategoryState('planets');
                    return (
                      <CategoryContainer
                        toggleFavorite={this.toggleFavorite}
                        currentData={planetsData}
                      />
                    );
                  }}
                />
                <Route
                  exact
                  path="/vehicles"
                  render={() => {
                    this.setCategoryState('vehicles');
                    return (
                      <CategoryContainer
                        toggleFavorite={this.toggleFavorite}
                        currentData={vehiclesData}
                      />
                    );
                  }}
                />
              </Switch>
            </main>
          </div>
        </Router>
        <footer>
          <ScrollText />
        </footer>
      </div>
    );
  }
}

export default App;
