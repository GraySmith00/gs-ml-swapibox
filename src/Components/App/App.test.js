import React from 'react';
import ReactDOM from 'react-dom';
import { shallow } from 'enzyme';
import App from '../App';

describe('App component', () => {
  let wrapper;

  beforeEach(() => {
    wrapper = shallow(<App />);
  });
  it.skip('renders without crashing', () => {
    const div = document.createElement('div');
    ReactDOM.render(<App />, div);
    ReactDOM.unmountComponentAtNode(div);
  });

  it.skip('matches the snapshot', () => {
    expect(wrapper.html()).toMatchSnapshot();
  });
});
