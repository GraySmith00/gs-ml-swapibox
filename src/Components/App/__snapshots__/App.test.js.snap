// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`App component router should match Snapshot of landing page 1`] = `undefined`;

exports[`App component router should match Snapshot of people page 1`] = `undefined`;

exports[`App component router should match Snapshot of planets page 1`] = `undefined`;

exports[`App component router should match Snapshot of vehicles page 1`] = `undefined`;

exports[`App component snapshot should match the snapshot 1`] = `undefined`;
